jgrapht-core/src/main/java/org/jgrapht/alg/isomorphism/ColorRefinementIsomorphismInspector.java

```diff
@@ -135,17 +189,21 @@ public boolean isomorphismExists() {
             return false;
         }
 
-        ColorRefinementAlgorithm<V, E> colorRefinementAlgorithm1 = new ColorRefinementAlgorithm<>(graph1);
-        ColorRefinementAlgorithm<V, E> colorRefinementAlgorithm2 = new ColorRefinementAlgorithm<>(graph2);
+        Graph<DistinctGraphVertex<V, E>, DistinctGraphEdge<V, E>> graph = getDisjointGraphUnion(graph1, graph2);
+        ColorRefinementAlgorithm<DistinctGraphVertex<V, E>, DistinctGraphEdge<V, E>> colorRefinementAlgorithm =
+                new ColorRefinementAlgorithm<>(graph);
 
-        // execute color refinement for graph1
-        Coloring<V> coloring1 = colorRefinementAlgorithm1.getColoring();
-        // execute color refinement for graph2
-        Coloring<V> coloring2 = colorRefinementAlgorithm2.getColoring();
+        // execute color refinement for graph
+        Coloring<DistinctGraphVertex<V, E>> coloring = colorRefinementAlgorithm.getColoring();
 
         isomorphismTestExecuted = true;
 
-        isIsomorphic = coarseColoringAreEqual(coloring1, coloring2);
+        isIsomorphic = coarseColoringAreEqual(coloring);
+
+        if(isIsomorphic && !assertIsomorphism(graph1, graph2, (IsomorphicGraphMapping<V, E>) isomorphicGraphMapping)) {
+            throw new IllegalStateException("Either there is no isomorphism or the mapping is incorrect!");
```

Comment:
Instead of if+throw, use a Java assert statement.  That way, in a production deployment, the configuration can be set up to skip the check for performance.